src/index.ts

```diff
@@ -434,15 +465,15 @@ class Spanner extends GrpcService {
         {
           name: formattedName,
           displayName: shortName,
+          nodeCount: config.nodes || 1,
```

Comment:
As also mentioned [here](https://github.com/googleapis/nodejs-spanner/pull/798#discussion_r369383379): I think we should flag in the release notes that users that have been using `config.nodeCount` must switch to using `config.nodes`. As `config.nodeCount` was only introduced by the adding of types relatively recently, and it is not mentioned in the documentation, I think it's fair to assume that it should not affect many users.

Reply:
To summarize,

- `nodeCount` was never sighted in any docs. 
- It erroneously became a part of the `config` when types where [introduced](https://github.com/googleapis/nodejs-spanner/blob/da8c2f85b55346d99a9eacc4249e6d6cd1a14556/src/instance.ts#L63) and `CreateInstanceRequest` interface extended `IInstance` interface
- This PR removes `nodeCount` and some other misleading params (`name`, `state`, `endpointUris`) from the `CreateInstance` `config` interface.
Bottom line, if anyone used `nodeCount` option, this change will break their code.

@bcoe what should be the proper course of action here?

Reply:
From our meeting notes:

On Node meeting @bcoe suggested to lump this together with other breaking changes in the pipeline anyways. Microgenerators with types are coming.